rdmo/projects/progress.py

```diff
@@ -47,6 +47,44 @@ def compute_sets(values):
     return sets
 
 
+def compute_next_relevant_page(current_page, direction, catalog, resolved_conditions):
+    # compute the next relevant page based on resolved conditions.
+    # If no current_page, return None
+    if not current_page:
+        return None
+
+    # Check if the current page meets the conditions
+    if compute_show_page(current_page, resolved_conditions):
```

Comment:
the first page was already checked when we enter the recursion, I think we should first find the next page, then check if `None`, then `compute_show_page`, then return the pae or call the function again.